apps/admin-gui/src/app/facilities/pages/facility-select-page/facility-select-page.component.ts

```diff
@@ -60,9 +64,14 @@ export class FacilitySelectPageComponent implements OnInit {
 
     const dialogRef = this.dialog.open(CreateFacilityDialogComponent, config);
 
-    dialogRef.afterClosed().subscribe(result => {
-      if (result) {
-        this.refreshTable();
+    dialogRef.afterClosed().subscribe(facilityCreated => {
+      if (facilityCreated) {
+        this.loading = true;
+        this.authzResolver.getPerunPrincipal().subscribe(perunPrincipal => {
+          this.storeService.setPerunPrincipal(perunPrincipal);
+          this.guiAuthResolver.init(perunPrincipal);
+          this.refreshTable();
+        }, () => this.loading = false);
```

Comment:
```suggestion
        this.loading = true;
        this.initAuthService.loadPrincipal().then(() => this.refreshTable());
```
you can remove console log from loadPrincipal method, please use this method in vos and groups too